client/packages/common/src/ui/layout/tables/material-react-table/useBaseMaterialTable.tsx

```diff
@@ -51,25 +50,47 @@ export const useBaseMaterialTable = <T extends MRT_RowData>({
   const { getTableLocalisations } = useIntlUtils();
   const localization = getTableLocalisations();
 
-  const { columns, defaultHiddenColumns, defaultColumnPinning } =
-    useMaterialTableColumns(omsColumns);
+  const { columns } = useMaterialTableColumns(omsColumns);
 
   // Filter needs to be applied after columns are processed
   const { columnFilters, onColumnFiltersChange } = useTableFiltering(columns);
   const { sorting, onSortingChange } = useUrlSortManagement(initialSort);
 
-  const initialState = useRef(
-    getSavedTableState<T>(tableId, defaultHiddenColumns, defaultColumnPinning)
-  );
-  const [columnOrder, setColumnOrder] = useState(
-    initialState.current.columnOrder ?? []
-  );
-
   const processedData = useMemo(
     () => getGroupedRows(data, groupByField, t),
     [data, groupByField]
   );
 
+  const density = useColumnDensity(tableId);
+  const columnSizing = useColumnSizing(tableId);
+  const columnVisibility = useColumnVisibility(tableId, columns);
+  const columnPinning = useColumnPinning(tableId, columns);
+  const columnOrder = useColumnOrder(
+    tableId,
+    columns,
+    enableRowSelection,
+    enableColumnResizing,
+    groupByField
+  );
+
+  const resetTableState = () => {
+    table.setColumnVisibility(columnVisibility.initial);
+    table.setDensity(density.initial);
+
+    table.resetColumnOrder();
+    table.resetColumnPinning();
+    table.resetColumnSizing();
+    table.resetSorting();
+    table.resetColumnFilters();
```

Comment:
There actually is a `table.reset()` function, which resets all state. Unfortunately, this doesn't fire the individual onChange handlers, which we need in order to update our now manually managed state. So have to call each individually

Reply:
I had a play around with clearing the local storage as part of the updaters, and I can see why you've done it the way you have now (particularly with column visibility), so it's not so easy to clear it from the individual updaters, so I'm tempted to just chuck this in here:
```ts
setTimeout(
  () => localStorage.removeItem(`@openmsupply-client/tables/${tableId}`),
  500
);
```

Or put that in utils with the other localStorage methods :)